dataregistry-extension/src/notebooks.ts

```diff
@@ -4,31 +4,41 @@ import {
 } from "@jupyterlab/application";
 import { ICellModel, isCodeCellModel } from "@jupyterlab/cells";
 import {
+  createConverter,
   DataTypeNoArgs,
   DataTypeStringArg,
   nestedDataType,
   Registry,
   resolveDataType,
   TypedConverter,
-  createConverter
+  URLTemplate
 } from "@jupyterlab/dataregistry";
 import { IOutputModel } from "@jupyterlab/rendermime";
 import { ReadonlyJSONObject, ReadonlyJSONValue } from "@phosphor/coreutils";
 import { defer, Observable, of } from "rxjs";
 import { map, switchMap } from "rxjs/operators";
+import { notebookContextDataType } from "./documents";
 import {
   observableListToObservable,
   outputAreaModelToObservable
 } from "./observables";
 import { IRegistry } from "./registry";
-import { notebookContextDataType } from "./documents";
 
-/**
- * This defines a nested data type for notebooks, so that a notebook
- * `file:///notebook.ipynb` will have children `file:///notebook.ipynb#/cells/0/outputs/0`,etc
- */
+const notebookURL = new URLTemplate("file://{+path}", {
```

Comment:
@ellisonbg If you are interested, this  is how it looks now using URL templates. I wrap them in a class so we can use  them in type safe way, by specifying  the  types of the  different args.